backend/core/database.py

```diff
@@ -1209,17 +1245,83 @@ def handle_hijack_multiple_action(self, message):
+                                ):
+                                    log.debug(
+                                        "deleting hijack using cache for bgp updates"
+                                    )
+                                    redis_hijack = yaml.safe_load(redis_hijack)
+                                    log.debug(
+                                        "bgpupdate_keys {} for {}".format(
+                                            redis_hijack["bgpupdate_keys"], redis_hijack
+                                        )
+                                    )
+                                    self.wo_db.execute(
+                                        "DELETE FROM bgp_updates WHERE %s = ANY(hijack_key) AND handled = true AND array_length(hijack_key,1) = 1 AND key = ANY(%s);",
+                                        (
+                                            hijack_key,
+                                            list(redis_hijack["bgpupdate_keys"]),
+                                        ),
+                                    )
+                                    self.wo_db.execute(
+                                        "UPDATE bgp_updates SET hijack_key = array_remove(hijack_key, %s) WHERE handled = true AND key = ANY(%s);",
+                                        (
+                                            hijack_key,
+                                            list(redis_hijack["bgpupdate_keys"]),
+                                        ),
+                                    )
+                                else:
+                                    log.debug(
+                                        "deleting hijack by querying bgp updates database"
+                                    )
+                                    self.wo_db.execute(
+                                        "DELETE FROM bgp_updates WHERE %s = ANY(hijack_key) AND array_length(hijack_key,1) = 1 AND handled = true;",
+                                        (hijack_key,),
+                                    )
+                                    self.wo_db.execute(
+                                        "UPDATE bgp_updates SET hijack_key = array_remove(hijack_key, %s) WHERE %s = ANY(hijack_key) AND handled = true;",
+                                        (hijack_key, hijack_key),
+                                    )
+                                    log.debug(
+                                        "bgpupdate_keys is empty for {}".format(
+                                            redis_hijack
+                                        )
+                                    )
                             else:
                                 raise BaseException("unreachable code reached")
 
-                    except Exception:
+                    except Exception as e:
                         log.exception("{}".format(raw))
+                        self.producer.publish(
```

Comment:
why didn't we have this before, could you add a short comment why it was needed and what it does?